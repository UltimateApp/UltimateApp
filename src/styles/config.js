export default {
  swipeConfig: {
    velocityThreshold: 0.2,
    directionalOffsetThreshold: 80,
  },
};
