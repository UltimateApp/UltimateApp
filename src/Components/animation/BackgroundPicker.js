import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Picker } from '@react-native-community/picker';

import theme from '../../styles/theme.style';
import I18n from '../../utils/i18n';
import { AnimationBackgrounds } from '../../Fixtures/config';

const BackgroundPicker = (props) => {
  return (
    <Picker
      selectedValue={props.selectedBackground}
      style={styles.picker}
      itemStyle={styles.itemStyle}
      onValueChange={(itemValue, itemIndex) => {
        props.onBackgroundChange(itemValue);
      }}
    >
      {Object.values(AnimationBackgrounds).map((background) => (
        <Picker.Item label={I18n.t(`data.animationBackgrounds.${background}`)} value={background} key={background} />
      ))}
    </Picker>
  );
};

export default BackgroundPicker;

const styles = StyleSheet.create({
  picker: {
    height: 80,
    width: 150,
  },
  itemStyle: {
    fontSize: theme.FONT_SIZE_SMALL,
    height: 80,
  },
});
