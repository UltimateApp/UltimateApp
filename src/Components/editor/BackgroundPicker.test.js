import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { AnimationBackgrounds } from '../../Fixtures/config';

import BackgroundPicker from './BackgroundPicker';

describe('<BackgroundPicker />', () => {
  afterEach(() => jest.clearAllMocks());

  const onBackgroundChange = jest.fn();
  const selectedBackground = 'endzone';

  it('renders correctly', async () => {
    const { toJSON } = await waitFor(() =>
      render(<BackgroundPicker onBackgroundChange={onBackgroundChange} selectedBackground={selectedBackground} />),
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
