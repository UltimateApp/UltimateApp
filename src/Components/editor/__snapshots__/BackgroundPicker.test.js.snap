// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundPicker /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": 80,
      "width": 120,
    }
  }
>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "Rectangle",
          "textColor": undefined,
          "value": "rectangle",
        },
        Object {
          "label": "Endzone",
          "textColor": undefined,
          "value": "endzone",
        },
        Object {
          "label": "3/4 field",
          "textColor": undefined,
          "value": "three quarters field",
        },
        Object {
          "label": "Empty",
          "textColor": undefined,
          "value": "empty",
        },
      ]
    }
    onChange={[Function]}
    selectedIndex={1}
    style={
      Array [
        Object {
          "height": 216,
        },
        Object {
          "fontSize": 14,
          "height": 80,
        },
      ]
    }
  />
</View>
`;
